### PR TITLE
Backports #15853 #15463 and #16073

### PR DESCRIPTION
**Pull request by pravblockc**
Backports #15853 #15463 and #16073

### CODE REVIEW

File: test/functional/rpc_psbt.py [hunk 1]
```diff
@@ -87,10 +87,11 @@ def run_test(self):
         new_psbt = self.nodes[0].converttopsbt(rawtx['hex'])
         self.nodes[0].decodepsbt(new_psbt)
 
-        # Make sure that a psbt with signatures cannot be converted
+        # Make sure that a non-psbt with signatures cannot be converted
+        # Error could be either "TX decode failed" (segwit inputs causes parsing to fail) or "Inputs must not have scriptSigs and scriptWitnesses"
```
Comment (PastaPastaPasta) [inline test/functional/rpc_psbt.py:91]: segwit references

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
one things, otherwise looks good if CI is happy

File: test/functional/rpc_psbt.py (hunk 1 above)
Comment (pravblockc) [reply test/functional/rpc_psbt.py:91]: since Dash don't use segwit, do you the PR needs to be backported? The error message is valid here

Comment (PastaPastaPasta) [reply test/functional/rpc_psbt.py:91]: yeah, should just drop this

Comment (pravblockc) [reply test/functional/rpc_psbt.py]: PR #14818 has been dropped(not valid)

**Review by PastaPastaPasta — DISMISSED** [review]
utACK for merging via merge commit

File: src/wallet/rpcwallet.cpp [hunk 2]
```diff
@@ -3775,9 +3775,20 @@ static UniValue getaddressesbylabel(const JSONRPCRequest& request)
 
     // Find all addresses that have the given label
     UniValue ret(UniValue::VOBJ);
-    for (const std::pair<CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {
+    std::set<std::string> addresses;
+    for (const std::pair<const CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {
```
Comment (UdjinM6) [inline src/wallet/rpcwallet.cpp:3779]: 15463: unrelated, should keep `for (const std::pair<CTxDestination, CAddressBookData>& item : pwallet->mapAddressBook) {` here.

NOTE: we should backport missing parts of 13241.

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/wallet/rpcwallet.cpp (hunk 2 above)
Comment (pravblockc) [reply src/wallet/rpcwallet.cpp]: agree and updated code.

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**